website/addons/owncloud/utils.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+from website.oauth.models import ExternalAccount
+from website.security import encrypt, decrypt
+
+
+class ExternalAccountConverter:
```

Comment:
The behavior that this class provides isn't necessarily something unique to ownCloud -- it's something that would need to be done for any service provider that uses Basic Auth. Consider moving this to somewhere like the `website.oauth.models` module, renaming it `BasicAuthProviderMixin`, and having `OwnCloudProvider` inherit from it.

(And updating the docstring)